src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -966,6 +971,86 @@ public ByteArrayInputStream getThumbnail(long datasetId, long imageId) {
+   * @param instrument containing information to be associated with the image
+   * @param imageId ID of the image that will be annotated
+   */
+  public void associateInstrumentWithImage(Instrument instrument, Long imageId) {
+    if (!isConnected()) {
+      this.connect();
+    }
+    // 1. Check if an image with the id is in OMERO. Throw a runtime exception if not
+    ImageData imageData;
+    try {
+      BrowseFacility browseFacility = gateway.getFacility(BrowseFacility.class);
+      imageData = browseFacility.getImage(securityContext, imageId);
+    } catch (DSOutOfServiceException dsOutOfServiceException) {
+      throw new RuntimeException("Error while accessing omero service: broken connection, expired session or not logged in", dsOutOfServiceException);
+    } catch (ExecutionException executionException) {
+      throw new RuntimeException("Task aborted unexpectedly.", executionException);
+    } catch (DSAccessException dsAccessException) {
+      throw new RuntimeException("Could not pull data from the omero server.", dsAccessException);
+    }
+
+    if (imageData == null) {
+      throw new IllegalArgumentException("No image available with image id " + imageId);
+    }
+    // 2. Use the instrument metadata and store them accordingly next to the image in OMERO.
+    List<NamedValue> mapAnnotationContent = new ArrayList<>();
+    ImagingHardware instrumentHardware = instrument.getHardware();
+    Location instrumentLocation = instrument.getLocation();
+    Address instrumentLocationAddress = instrumentLocation.getAddress();
+    try {
+      mapAnnotationContent.add(new NamedValue("hardware.detector.type", instrumentHardware.getDetector().getType()));
+    } catch (NullPointerException ignored) {}
+    try {
+      mapAnnotationContent.add(new NamedValue("hardware.objective", instrumentHardware.getObjective()));
+    } catch (NullPointerException ignored) {}
+    try {
+      mapAnnotationContent.add(new NamedValue("manufacturer", instrument.getManufacturer()));
+    } catch (NullPointerException ignored) {}
+    try {
+      mapAnnotationContent.add(new NamedValue("location.address.affiliation", instrumentLocationAddress.getAffiliation()));
+    } catch (NullPointerException ignored) {}
+    try {
+      mapAnnotationContent.add(new NamedValue("location.address.country", instrumentLocationAddress.getCountry()));
+    } catch (NullPointerException ignored) {}
+    try {
+      mapAnnotationContent.add(new NamedValue("location.address.street", instrumentLocationAddress.getStreet()));
+    } catch (NullPointerException ignored) {}
+    try {
+      mapAnnotationContent.add(new NamedValue("location.address.zipCode", instrumentLocationAddress.getZipCode().toString()));
+    } catch (NullPointerException ignored) {}
+    try {
+      mapAnnotationContent.add(new NamedValue("location.roomId", instrumentLocation.getRoomId()));
+    } catch (NullPointerException ignored) {}
```

Comment:
Suggestion: the underlying problem is, that the instrument properties can be null. So lets make sure they are never null, but empty Strings, Lists or empty objects.

Reply:
this needs to be addressed in the instrument constructor then.

Reply:
I created an issue in the data-model-lib qbicsoftware/data-model-lib#55